test/EndToEndTests/Tests/Client/Microsoft.OData.Client.E2E.Tests/ClientTests/Server/ClientQueryTestsController.cs

```diff
@@ -0,0 +1,167 @@
+﻿//-----------------------------------------------------------------------------
+// <copyright file="ClientQueryTestsController.cs" company=".NET Foundation">
+//      Copyright (c) .NET Foundation and Contributors. All rights reserved.
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.OData.Formatter;
+using Microsoft.AspNetCore.OData.Query;
+using Microsoft.AspNetCore.OData.Routing.Controllers;
+using Microsoft.OData.Client.E2E.Tests.Common.Server.EndToEnd;
+
+namespace Microsoft.OData.Client.E2E.Tests.ClientTests.Server
+{
+    public class ClientQueryTestsController : ODataController
+    {
+        [EnableQuery]
+        [HttpGet("odata/People")]
+        public IActionResult Get()
+        {
+            var people = CommonEndToEndDataSource.People;
+
+            return Ok(people);
+        }
+
+        [EnableQuery]
+        [HttpGet("odata/People({key})")]
+        public IActionResult Get(int key)
+        {
+            var person = CommonEndToEndDataSource.People?.FirstOrDefault(a => a.PersonId == key);
+
+            if (person == null)
+            {
+                return NotFound();
+            }
+
+            return Ok(person);
+        }
+
+        [EnableQuery]
+        [HttpGet("odata/Logins")]
+        public IActionResult GetLogins()
+        {
+            var logins = CommonEndToEndDataSource.Logins;
+
+            return Ok(logins);
+        }
+
+        [EnableQuery]
+        [HttpGet("odata/Logins({key})")]
+        public IActionResult GetLogin(string key)
+        {
+            var login = CommonEndToEndDataSource.Logins?.FirstOrDefault(a => a.Username == key);
+            
+            if (login == null)
+            {
+                return NotFound();
+            }
+
+            return Ok(login);
+        }
+
+        [EnableQuery]
+        [HttpGet("odata/Logins({key})/SentMessages(FromUsername={FromUsername},MessageId={MessageId})")]
+        public IActionResult GetLoginSentMessages([FromODataUri] string key, [FromODataUri] string FromUsername, [FromODataUri] int MessageId)

```

Comment:
```suggestion
        [HttpGet("odata/Logins({key})/SentMessages(fromUsername={fromUsername},messageId={messageId})")]
        public IActionResult GetLoginSentMessages([FromODataUri] string key, [FromODataUri] string fromUsername, [FromODataUri] int messageId)
```

Reply:
The first `fromUserName` is the property name.. Should it be in PascalCasing just as it is in the model class or the casing doesn't matter